File: lib/models/tutor.dart
// ignore_for_file: non_constant_identifier_names

class Tutor {
  String? tutor_id;
  String? tutor_email;
  String? tutor_phone;
  String? tutor_name;
  String? tutor_password;
  String? tutor_description;
  String? tutor_subject;

  Tutor({
    this.tutor_id,
    this.tutor_email,
    this.tutor_phone,
    this.tutor_name,
    this.tutor_password,
    this.tutor_description,
    this.tutor_subject,
  });

  Tutor.fromJson(Map<String, dynamic> json) {
    tutor_id = json['tutor_id'].toString();
    tutor_email = json['tutor_email'].toString();
    tutor_phone = json['tutor_phone'].toString();
    tutor_name = json['tutor_name'].toString();
    tutor_password = json['tutor_password,'].toString();
    tutor_description = json['tutor_description'].toString();
    tutor_subject = json['tutor_subject'].toString();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['tutor_id'] = tutor_id;
    data['tutor_email'] = tutor_email;
    data['tutor_phone'] = tutor_phone;
    data['tutor_name'] = tutor_name;
    data['tutor_password,'] = tutor_password;
    data['tutor_description'] = tutor_description;
    data['tutor_subject'] = tutor_subject;
    return data;
  }
}

File: php/tutor.php
<?php
include("dbconnect.php");
if (!isset($_POST)) {
    $response = array('status' => 'failed', 'data' => null);
    sendJsonResponse($response);
    die();
}

$limit = 5;
$page = (isset($_POST['page']) && is_numeric($_POST['page'])) ? $_POST['page'] : 1;
$pageStart = ($page - 1) * $limit;
$sqlselectTutor = "SELECT * FROM tbl_tutors LIMIT $pageStart, $limit";
$stmt = $conn->prepare($sqlselectTutor);
$stmt->execute();
$result = $stmt->get_result();

$sql = $conn->query("SELECT count(tutor_id) AS id FROM tbl_tutors")->fetch_assoc();
$allRecords = $sql['id'];
$totalPages = ceil($allRecords / $limit);
$prev = $page - 1;
$next = $page + 1;

if ($allRecords > 0) {
    $tutors['tutors'] = array();
    while ($row = $result->fetch_assoc()) {
        $tutlist = array();
        $tutlist['tutor_id'] = $row['tutor_id'];
        $tutlist['tutor_email'] = $row['tutor_email'];
        $tutlist['tutor_phone'] = $row['tutor_phone'];
        $tutlist['tutor_name'] = $row['tutor_name'];
        $tutlist['tutor_description'] = $row['tutor_description'];
        array_push($tutors['tutors'], $tutlist);
    }
    $response = array('status' => 'success', 'page' => "$page", 'totalPages' => "$totalPages", 'data' => $tutors);
    sendJsonResponse($response);
} else {
    $response = array('status' => 'failed', 'page' => "$page", 'totalPages' => "$totalPages", 'data' => null);
    sendJsonResponse($response);
}


function sendJsonResponse($sentArray)
{
    header('Content-Type: application/json');
    echo json_encode($sentArray);
}
?>